jobs/payment-jobs/tasks/cfs_create_invoice_task.py

```diff
@@ -332,6 +334,132 @@ def _create_pad_invoices(cls):  # pylint: disable=too-many-locals
+            for invoice in account_invoices:
+                lines.extend(invoice.payment_line_items)
+                invoice_total += invoice.total
+
+            invoice_number = account_invoices[-1].id
+            try:
+                # Get the first invoice id as the trx number for CFS
+                invoice_response = CFSService.create_account_invoice(transaction_number=invoice_number,
+                                                                     line_items=lines,
+                                                                     cfs_account=cfs_account)
+            except Exception as e:  # NOQA # pylint: disable=broad-except
+                # There is a chance that the error is a timeout from CAS side,
+                # so to make sure we are not missing any data, make a GET call for the invoice we tried to create
+                # and use it if it got created.
+                current_app.logger.info(e)  # INFO is intentional as sentry alerted only after the following try/catch
+                has_invoice_created: bool = False
+                try:
+                    # add a 10 seconds delay here as safe bet, as CFS takes time to create the invoice
+                    time.sleep(10)
+                    invoice_number = generate_transaction_number(str(invoice_number))
+                    invoice_response = CFSService.get_invoice(
+                        cfs_account=cfs_account, inv_number=invoice_number
+                    )
+                    has_invoice_created = invoice_response.get('invoice_number', None) == invoice_number
+                    invoice_total_matches = Decimal(invoice_response.get('total', '0')) == invoice_total
+                except Exception as exc:  # NOQA # pylint: disable=broad-except,unused-variable
+                    # Ignore this error, as it is irrelevant and error on outer level is relevant.
+                    pass
+                # If no invoice is created raise an error for sentry
+                if not has_invoice_created:
+                    capture_message(f'Error on creating EFT invoice: account id={payment_account.id}, '
+                                    f'auth account : {payment_account.auth_account_id}, ERROR : {str(e)}',
+                                    level='error')
+                    current_app.logger.error(e)
+                    continue
+                if not invoice_total_matches:
+                    capture_message(f'Error on creating EFT invoice: account id={payment_account.id}, '
+                                    f'auth account : {payment_account.auth_account_id}, Invoice exists: '
+                                    f' CAS total: {invoice_response.get("total", 0)}, PAY-BC total: {invoice_total}',
+                                    level='error')
+                    current_app.logger.error(e)
+                    continue
+
+            mailer.publish_mailer_events(MailerEvents.EFT_INVOICE_CREATED, payment_account, {
```

Comment:
while you're in there, see if you can fix these too:

![image](https://github.com/bcgov/sbc-pay/assets/3484109/cf12a51c-894f-4edb-898a-0d112773bda1)

Reply:
![image](https://github.com/bcgov/sbc-pay/assets/3484109/1542c778-b2b5-4715-bbb8-0a3ccd366e93)

Reply:
add to here: sbc-pay/pay-api/src/pay_api/utils/enums.py